packages/discord.js/src/client/actions/ThreadMembersUpdate.js

```diff
@@ -8,24 +9,35 @@ class ThreadMembersUpdateAction extends Action {
     const client = this.client;
     const thread = client.channels.cache.get(data.id);
     if (thread) {
-      const old = thread.members.cache.clone();
       thread.memberCount = data.member_count;
+      const addedMembers = new Collection();
+      const removedMembers = new Collection();
 
-      data.added_members?.forEach(rawMember => {
-        thread.members._add(rawMember);
-      });
+      data.added_members?.reduce(
+        (_addedMembers, addedMember) => _addedMembers.set(addedMember.user_id, thread.members._add(addedMember)),
+        addedMembers,
+      );
 
-      data.removed_member_ids?.forEach(memberId => {
-        thread.members.cache.delete(memberId);
-      });
+      data.removed_member_ids?.reduce((removedMembersIds, removedMembersId) => {
+        const cachedThreadMember = thread.members.cache.get(removedMembersId);
+        if (cachedThreadMember) removedMembersIds.set(cachedThreadMember.id, cachedThreadMember);
```

Comment:
I think something allowing you to know who left regardless of whether the thread member was cached would be nice.
(Partials or just a `null` value maybe?)

Reply:
Yes, we should at very least give some data back, probably a PartialUser so we can read the IDs when iterating the maps with `.values()`

Reply:
I've gone ahead and created partial thread members. This should resolve the uncached behaviour here.